api/envoy/config/route/v3/route_components.proto

```diff
@@ -2198,8 +2198,29 @@ message RateLimit {
       option (udpa.annotations.versioning).previous_message_type =
           "envoy.api.v2.route.RateLimit.Action.GenericKey";
 
-      // The value to use in the descriptor entry.
-      string descriptor_value = 1 [(validate.rules).string = {min_len: 1}];
+      oneof descriptor_specifier {
+        // The value to use in the descriptor entry.
+        string descriptor_value = 1 [(validate.rules).string = {min_len: 1}];
+
+        // Substitution format string to generate the value for the descriptor entry.
+        // The same :ref:`format specifier <config_access_log_format>` as used for
+        // :ref:`HTTP access logging <config_access_log>` applies here.
+        //
+        // .. note::
+        //
+        //   The format string must contain exactly one valid substitution field. If the format string
+        //   does not meet this requirement, the configuration will be rejected.
+        //
+        //   If the format string evaluation fails during request processing, this descriptor will be 
+        //   skipped and not included in the rate limit call.
+        //
+        // For example, ``%REQ(:method)%`` will be replaced with the HTTP method, and
+        // ``%CEL(request.headers['user-id'] | 'unknown')%`` will use CEL to extract the user ID
+        // from request headers with a fallback value.
+        //
+        // Only one of ``descriptor_value`` or ``descriptor_format`` should be set.
+        string descriptor_format = 3 [(validate.rules).string = {min_len: 1}];
+      }
```

Comment:
This is break change to API and we won't suggest to use the oneof now. You can add a new field directly and remove the PGV validation (`[(validate.rules).string = {min_len: 1}]`) at the field. All necessary API validation then could be done in your code.